src/main/java/com/minecolonies/coremod/entity/ai/mobs/EntityAIAttackRangedBowBarbarian.java

```diff
@@ -0,0 +1,196 @@
+        this.entity.setSwingingArms(true);
+    }
+
+    /**
+     * Resets the task
+     */
+    public void resetTask()
+    {
+        super.resetTask();
+        this.entity.setSwingingArms(false);
+        this.seeTime = 0;
+        this.attackTime = -1;
+        this.entity.resetActiveHand();
+    }
+
+    /**
+     * Updates the task
+     */
+    public void updateTask()
+    {
+        final EntityLivingBase entitylivingbase = this.entity.getAttackTarget();
+
+        if (entitylivingbase != null)
+        {
+            final double d0 = this.entity.getDistanceSq(entitylivingbase.posX, entitylivingbase.getEntityBoundingBox().minY, entitylivingbase.posZ);
+            final boolean flag = this.entity.getEntitySenses().canSee(entitylivingbase);
+            final boolean flag1 = this.seeTime > 0;
+
+            if (flag != flag1)
+            {
+                this.seeTime = 0;
+            }
+
+            if (flag)
+            {
+                ++this.seeTime;
+            }
+            else
+            {
+                --this.seeTime;
+            }
+
+            if (d0 <= (double) this.maxAttackDistance && this.seeTime >= TWENTY_SECONDS)
+            {
+                this.entity.getNavigator().clearPathEntity();
+                ++this.strafingTime;
+            }
+            else
+            {
+                this.entity.getNavigator().tryMoveToEntityLiving(entitylivingbase, this.moveSpeedAmp);
+                this.strafingTime = -1;
+            }
+
+            final int strafeTime = TWENTY_SECONDS;
+
+            if (this.strafingTime >= strafeTime)
+            {
+                final double maxRNG = 0.3D;
+
+                if ((double) this.entity.getRNG().nextFloat() < maxRNG)
+                {
+                    this.strafingClockwise = !this.strafingClockwise;
+                }
+
+                if ((double) this.entity.getRNG().nextFloat() < maxRNG)
+                {
+                    this.strafingBackwards = !this.strafingBackwards;
+                }
+
+                this.strafingTime = 0;
+            }
+
+            if (this.strafingTime > -1)
+            {
+                if (d0 > (this.maxAttackDistance * THREE_QUARTERS_BLOCK))
+                {
+                    this.strafingBackwards = false;
+                }
+                else if (d0 < (this.maxAttackDistance * ONE_QUARTER_BLOCK))
+                {
+                    this.strafingBackwards = true;
+                }
+
+                this.entity.getMoveHelper().strafe(this.strafingBackwards ? -HALF_BLOCK : HALF_BLOCK, this.strafingClockwise ? HALF_BLOCK : -HALF_BLOCK);
+                this.entity.faceEntity(entitylivingbase, MAX_DEGREES, MAX_DEGREES);
+            }
+            else
+            {
+                this.entity.getLookHelper().setLookPositionWithEntity(entitylivingbase, MAX_DEGREES, MAX_DEGREES);
+            }
+
+            final int alternateAttackTime = --this.attackTime;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this increment or decrement operator into a dedicated statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)